### PR TITLE
Add Examples of Collection Initializers

### PR DESCRIPTION
**Pull request by anonymousgituser**
# Add Examples of Collection Initializers

## Summary

1. Added example showing an Add method with multiple parameters and its use with a collection initializer.

2. Added example with Add method using a variable number of parameters along with the use of indexers.

## Details

This change adds two examples to expound upon collection initializers.

The first example demonstrates using multiple parameters in an Add method and how to initialize a collection with multiple entries per item.

In the second example, a variable number of parameters is used with an Add method of a collection. Along with this, the example includes an implementation of an indexer.

## Suggested Reviewers

Current contributors: @mairaw, @svick, @PatrickHofman, @guardrex, @tompratt-AQ, @mjhoffman65, @mikeblome

### CODE REVIEW

File: docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md [hunk 1]
```diff
@@ -76,9 +76,20 @@ var numbers = new Dictionary<int, string> {
...
   
-## Example  
+## Examples
+
+ The following example combines the concepts of object and collection initializers.
+
  [!code-csharp[csProgGuideLINQ#46](../../../csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_6.cs)]  
-  
+ 
+ Shown in the below example, an object which implements <xref:System.Collections.IEnumerable> containing an `Add` method with multiple parameters allows for collection initializers with multiple elements per item in the list corresponding to the signature of the `Add` method.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/classes-and-structs/object-and-collection-initializers.md]: For consistency, change "below example" to "following example".

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_7.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            return internalList.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return internalList.GetEnumerator();
+        }
+
+        public FormattedAddresses()
```
Comment (BillWagner) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_7.cs]: I'd prefer removing the constructor and initializing `internalList` as a field initializer.

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_7.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ExampleProject
+{
+    class FormattedAddresses : IEnumerable<string>
+    {
+        private List<string> internalList;
+        
+        public IEnumerator<string> GetEnumerator()
```
Comment (BillWagner) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_7.cs]: I'd prefer expression bodied members for both `GetEnumerator()` methods:

```csharp
public IEnumerator<string> GetEnumerator() => internalList.GetEnumerator();
```

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_8.cs [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ExampleProject
+{
+    class RudimentaryMultiValuedDictionary<TKey, TValue> : IEnumerable<KeyValuePair<TKey, List<TValue>>>
+    {
+        private Dictionary<TKey, List<TValue>> internalDictionary;
+
+        public IEnumerator<KeyValuePair<TKey, List<TValue>>> GetEnumerator()
```
Comment (BillWagner) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_8.cs]: Here again, I'd prefer expression bodied members.

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_8.cs [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        {
+            return internalDictionary.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return internalDictionary.GetEnumerator();
+        }
+
+        public RudimentaryMultiValuedDictionary()
```
Comment (BillWagner) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_8.cs]: As above, I'd prefer a field initializer and removing the default ctor.

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_8.cs [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        {
+            internalDictionary = new Dictionary<TKey, List<TValue>>();
+        }
+
+        public List<TValue> this[TKey key]
```
Comment (BillWagner) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_8.cs]: These could also both be expression bodied members.

File: docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_8.cs [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                return internalDictionary[key];
+            }
+            set
+            {
+                Add(key, value);
+            }
+        }
+
+        public void Add(TKey key, params TValue[] values)
+        {
```
Comment (BillWagner) [inline docs/csharp/programming-guide/arrays/codesnippet/CSharp/object-and-collection-initializers_8.cs]: Also an expression bodied member.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for adding these examples @anonymousgituser We appreciate it.

I had a few comments on the code. We're trying to promote the latest features where appropriate in our examples. I found several locations where expression bodied members would result in simpler code. (To that end, seeing the string interpolation in your samples made me :smile: ).

**Comment by anonymousgituser** [comment]
@BillWagner,

Changes made.

Thanks for the thorough and positive feedback! 😄

**Review by BillWagner — APPROVED** [review]

**Comment by BillWagner** [comment]
Thanks for making the updates @anonymousgituser I've approved this PR. 

Once the build finishes, we'll :shipit:.  The changes should show up on the live site in the next couple days, on our regular publish cycle.

Thanks again.

**Pull request merged**